### PR TITLE
user/topgrade: new package (15.0.0)

### PR DESCRIPTION
**Pull request by tulilirockz**

### CODE REVIEW

File: user/topgrade/template.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+pkgname = "topgrade"
+pkgver = "15.0.0"
+pkgrel = 0
+build_style = "cargo"
+hostmakedepends = ["cargo-auditable"]
+pkgdesc = "System updater written in Rust"
```
Comment (triallax) [inline user/topgrade/template.py:6]: like i said in irc, no "written in <lang>"

File: user/topgrade/template.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+def post_build(self):
+    TOPGRADE_BIN = f"target/{self.profile().triplet}/release/topgrade"
+    with open(f"{self.cwd}/topgrade.bash", "w") as o:
+        self.do(TOPGRADE_BIN, "--gen-completion", "bash", stdout=o)
+    with open(f"{self.cwd}/topgrade.fish", "w") as o:
+        self.do(TOPGRADE_BIN, "--gen-completion", "fish", stdout=o)
+    with open(f"{self.cwd}/_topgrade", "w") as o:
+        self.do(TOPGRADE_BIN, "--gen-completion", "zsh", stdout=o)
+    with open(f"{self.cwd}/topgrade.8", "w") as o:
+        self.do(TOPGRADE_BIN, "--gen-manpage", stdout=o)
```
Comment (triallax) [inline user/topgrade/template.py:23]:
```suggestion
    for shell in ["bash", "fish", "zsh"]:
        with open(f"{self.cwd}/topgrade.{shell}", "w") as o:
            self.do(f"target/{self.profile().triplet}/release/topgrade", "--gen-completion", "bash", stdout=o)
    with open(f"{self.cwd}/topgrade.8", "w") as o:
        self.do(f"target/{self.profile().triplet}/release/topgrade", "--gen-manpage", stdout=o)
```
also side note, we don't typically use screaming case in cports

File: user/topgrade/template.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def post_install(self):
+    self.install_completion("topgrade.bash", "bash")
+    self.install_completion("topgrade.fish", "fish")
+    self.install_completion("_topgrade", "zsh")
```
Comment (triallax) [inline user/topgrade/template.py:29]:
```suggestion
    self.install_completion("topgrade.zsh", "zsh")
```
per above

File: user/topgrade/template.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    self.install_man("topgrade.8")
+    self.install_file("config.example.toml", "usr/share/doc/topgrade")
```
Comment (triallax) [inline user/topgrade/template.py:31]: install to `/usr/share/examples/topgrade/config.toml` instead

**Review by triallax — CHANGES_REQUESTED** [review]

File: user/topgrade/template.py (hunk 3 above)
Comment (triallax) [reply user/topgrade/template.py:29]: or could even do a loop here as well

File: user/topgrade/template.py (hunk 2 above)
Comment (tulilirockz) [reply user/topgrade/template.py:23]: oh! i thought that ZSH *had* to have the `_` prefix! sorry for the suboptimal implementation

File: user/topgrade/template.py (hunk 1 above)
Comment (tulilirockz) [reply user/topgrade/template.py:6]: Fair!

File: user/topgrade/template.py (hunk 2 above)
Comment (nekopsykose) [reply user/topgrade/template.py]: it does, but install_completion renames the file

**Comment by tulilirockz** [comment]
Should be good!

**Pull request merged**